addons/mail/static/src/views/fields/statusbar_duration/statusbar_duration_field.js

```diff
@@ -0,0 +1,114 @@
+/** @odoo-module **/
+
+import { _lt } from "@web/core/l10n/translation";
+import {
+    preloadStatusBar,
+    statusBarField,
+    StatusBarField,
+} from "@web/views/fields/statusbar/statusbar_field";
+import { registry } from "@web/core/registry";
+
+export class StatusBarDurationField extends StatusBarField {
+    static template = "mail.StatusBarDurationField";
+
+    humanizeDuration(seconds, showFullDuration) {
+        if (seconds < 60) {
+            seconds = 60;
+        }
+        const minuteInSeconds = 60;
+        const hourInSeconds = 60 * minuteInSeconds;
+        const dayInSeconds = 24 * hourInSeconds;
+        const monthInSeconds = 30 * dayInSeconds;
+        const yearInSeconds = 365 * dayInSeconds;
+        const numberOfValues = showFullDuration ? 2 : 1;
+        const displayStyle = showFullDuration ? "long" : "narrow";
+
+        const years = Math.floor(seconds / yearInSeconds);
+        seconds -= years * yearInSeconds;
+        const months = Math.floor(seconds / monthInSeconds);
+        seconds -= months * monthInSeconds;
+        const days = Math.floor(seconds / dayInSeconds);
+        seconds -= days * dayInSeconds;
+        const hours = Math.floor(seconds / hourInSeconds);
+        seconds -= hours * hourInSeconds;
+        const minutes = Math.floor(seconds / minuteInSeconds);
+        seconds -= minutes * minuteInSeconds;
+        const durationObject = {};
+        if (years) {
+            durationObject.years = years;
+        }
+        if (months && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.months = months;
+        }
+        if (days && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.days = days;
+        }
+        if (hours && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.hours = hours;
+        }
+        if (minutes && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.minutes = minutes;
+        }
+        if (Object.keys(durationObject).length === 0) {
+            return false;
+        }
+        const duration = luxon.Duration.fromObject(durationObject);
```

Comment:
Some smart people found out a ``shiftTo`` function allowing to skip the whole split into sub-units, like
```
luxon.Duration.fromObject({seconds: mySeconds}).shiftTo("months", "dasy", "hours", "minutes", "seconds").toHuman();
```

Maybe it is worth investigating ? Not sure if we can skip having "0 years, 0 months, ..." in case of small numbers, did you play with that ?

Reply:
It works well with what you've sent me and I moved it in formatters with a docstring.

Since we only show 1 letter (1y, 1m, 1d, 1h, 1m), I still replace the "m" for months to "M" in English (or it's impossible to differentiate minutes and months) but I do in a shorter way.